File: javascript/collatz-conjecture/collatz-conjecture.js
export const steps = (num) => {
  if (num <= 0) {
    throw new Error("Only positive numbers are allowed");
  }
  let steps = 0;
  while (num != 1) {
    if (num % 2 == 0) {
      num = num / 2;
    } else {
      num = num * 3 + 1;
    }
    steps++;
  }
  return steps;
};

File: go/collatz-conjecture/collatz_conjecture.go
package collatzconjecture

import (
	"errors"
)

func CollatzConjecture(number int) (int, error) {
	if number <= 0 {
		return 0, errors.New("Only numbers above 0 are allowed")
	}
	var steps int
	for number != 1 {
		if number%2 == 0 {
			number = number / 2
		} else {
			number = number*3 + 1
		}
		steps++
	}
	return steps, nil
}
